pkg/codegen/dotnet/gen_test.go

```diff
@@ -15,21 +13,30 @@ import (
 )
 
 func TestGeneratePackage(t *testing.T) {
-	test.TestSDKCodegen(t, "dotnet", GeneratePackage, typeCheckGeneratedPackage)
+	test.TestSDKCodegen(t, &test.SDKCodegenOptions{
+		Language:   "dotnet",
+		GenPackage: generatePackageWithVersion,
+		Checks: map[string]test.CodegenCheck{
+			"dotnet/compile": typeCheckGeneratedPackage,
+		},
+	})
+}
+
+// TODO replace this with GeneratePackage when https://github.com/pulumi/pulumi/pull/7938 lands.
```

Comment:
https://github.com/pulumi/pulumi/pull/7938 has landed. It would be great to eliminate this, but we could also do it later.

Reply:
Done.